src/Raven.Server/Documents/Patch/PatchDocument.cs

```diff
@@ -267,9 +268,17 @@ protected PatcherOperationScope ApplySingleScript(DocumentsOperationContext cont
             }
         }
 
-        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine)
+        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine, string docId)
```

Comment:
docId should always come first from the document, then only if missing for external value (possible casing differences for example). If you will create the doc in PatchDocument.Apply then you do not need to pass the docId here I think.